tests/frontier/opcodes/test_callcode.py

```diff
@@ -0,0 +1,117 @@
+"""
+abstract: Tests the CALLCODE gas consumption with a CALL/CALLCODE value transfer.
+
+    Tests an EthereumJS bug where the CALLCODE gas was incorrectly calculated from a
+    CALL value transfer: https://github.com/ethereumjs/ethereumjs-monorepo/issues/3194
+
+    Test setup: 2 smart contract accounts exist where AAAB CALL/CALLCODEs AAAA, which then
+    CALLCODEs a non-existent contract AAAC. The result of AAAA's CALL/CALLCODE is stored in AAAB's
+    storage slot 0.
+"""
+
+from typing import Dict
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+    to_address,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+
+@pytest.fixture
+def caller_code(caller_gas, caller_type):
+    """
+    Code to call the callee contract:
+        PUSH1 0x00
+        DUP1 * 4
+        PUSH2 0xAAAA
+        PUSH2 call_gas
+        CALL/CALLCODE
+        PUSH1 0x00
+        SSTORE
+    """
+    return Op.SSTORE(0, caller_type(caller_gas, 0xAAAA, 0, 0, 0, 0, 0))
+
+
+@pytest.fixture
+def callee_code():
+    """
+    Code called by the caller contract:
+        PUSH1 0x00 * 4
+        PUSH1 0x00
+        PUSH1 0x00
+        PUSH2 0xAACC
+        PUSH2 0x1a90
```

Comment:
Could you add a comment on the value chosen here for gas? Sorry, if I've missed the obvious here.

Reply:
I don't think the value matters and is just arbitrary.

I believe we can copy over this: https://github.com/ethereumjs/ethereumjs-monorepo/pull/3214/files#diff-ca2747b296607c2b77ee28d7067ae931408fd2e023d421b588bed952d9cb035fR747

Reply:
If the value is arbitrary then I think using `GAS` is a better choice here.

Reply:
Agreed, to avoid confusion :smile: